src/plugins/intel_cpu/src/transformations/snippets/tpp/op/factory.cpp

```diff
@@ -0,0 +1,111 @@
+// partial specialization of the template for 1 and args. Note however that functional templates do not support partial specialization,
+// do we will need to create a dedicated class for that. Discuss if the templated solution is better on review.
+#define CREATE_UNARY_TPP_NODE(tpp_node_type) \
+    [](const std::shared_ptr<ov::Node>& ngraph_node) -> std::shared_ptr<ov::Node> { \
+        return std::make_shared<tpp_node_type>(ngraph_node->get_input_source_output(0)); \
+    }
+
+#define CREATE_BINARY_TPP_NODE(tpp_node_type) \
+    [](const std::shared_ptr<ov::Node>& ngraph_node) -> std::shared_ptr<ov::Node> { \
+        return std::make_shared<tpp_node_type>(ngraph_node->get_input_source_output(0), ngraph_node->get_input_source_output(1), ngraph_node->get_autob()); \
+    }
+
+#define CREATE_REDUCE_TPP_NODE(tpp_node_type) \
+    [](const std::shared_ptr<ov::Node>& ngraph_node) -> std::shared_ptr<ov::Node> { \
+        const auto& reduce = ov::as_type_ptr<snippets::op::ReduceBase>(ngraph_node); \
+        OPENVINO_ASSERT(reduce, "Attempt to create TPP Reduce from invalid node"); \
+        return std::make_shared<tpp_node_type>(reduce->get_input_source_output(0), reduce->get_axis()); \
+    }
+
+    std::unordered_map<ov::DiscreteTypeInfo, TPPNodeFactory::tpp_builder> TPPNodeFactory::m_direct_mappig {
+        {ov::op::v1::Add::get_type_info_static(), CREATE_BINARY_TPP_NODE(Add)},
+        {ov::op::v1::Subtract::get_type_info_static(), CREATE_BINARY_TPP_NODE(Subtract)},
+        {ov::op::v1::Multiply::get_type_info_static(), CREATE_BINARY_TPP_NODE(Multiply)},
+        {ov::op::v1::Divide::get_type_info_static(), CREATE_BINARY_TPP_NODE(Divide)},
+        {ov::op::v0::Exp::get_type_info_static(), CREATE_UNARY_TPP_NODE(Exp)},
+        {ov::op::v0::Relu::get_type_info_static(), CREATE_UNARY_TPP_NODE(Relu)},
+        // Note that we don't support conversion from ngraph ops here, since they have a broader semantics (e.g. multiple axis provided at a secont input)
+        {ov::snippets::op::ReduceMax::get_type_info_static(), CREATE_REDUCE_TPP_NODE(ReduceMax)},
+        {ov::snippets::op::ReduceSum::get_type_info_static(), CREATE_REDUCE_TPP_NODE(ReduceSum)},
+    };
+
+
+    std::vector<TPPNodeFactory::TPPCustomBuilder> TPPNodeFactory::m_custom_mapping{CustomPowerStaticBuilder()};
+    std::shared_ptr<ov::Node> TPPNodeFactory::create(const std::shared_ptr<ov::Node>& n) {
+        const auto& found = m_direct_mappig.find(n->get_type_info());
+        std::shared_ptr<ov::Node> tpp_node{nullptr};
+        if (found != m_direct_mappig.end()) {
+            tpp_node = (found->second)(n);
+        } else {
+            for (const auto& custom_builder : m_custom_mapping) {
+                if (custom_builder.matcher(n)) {
+                    tpp_node = custom_builder.builder(n);
+                    break;
+                }
+            }
+        }
+        if (tpp_node)
+            tpp_node->set_friendly_name(n->get_friendly_name());
```

Comment:
I am not sure that the factory should do that: it is usually done by the conversion transformations. Maybe we can move this logic there? What do you think?

Reply:
It actually depends on the convention we make. If we agree that all TPP's must have the same names as original ngraph ops, then this logic should be moved to the factory to avoid duplication and guarantee the conformance.

Honestly, I don't think this is super-important right now. Let's use the `replace_node_update_name` and remove it from the factory, I don't mind.